api-parent/access-api/src/main/java/com/sap/cloud/environment/servicebinding/api/ServiceBindingMerger.java

```diff
@@ -56,9 +74,23 @@ public List<ServiceBinding> getServiceBindings()
         return existingBindings.stream().anyMatch(contained -> equalityComparer.areEqual(contained, newBinding));
     }
 
+    /**
+     * Represents a method object that is capable of comparing two instances of {@link ServiceBinding}. An instance of
+     * this interface will be used when checking whether a given {@link ServiceBinding} is already contained in the
+     * combined set of {@link ServiceBinding}s.
+     */
     @FunctionalInterface
     public interface EqualityComparer
```

Comment:
I was looking for something "more standard" and therefore also thought about the `Comparator` interface.
However, the `Comparator` is supposed to impose a _total ordering_ of elements.
Therefore, it doesn't semantically fit here since we are not interested in ordering elements but rather to decide whether an object should be added to a `Set` or not.
I'm sure there are custom implementations of `Set` that allow defining some function for checking equality, but I don't want to introduce a new dependency just for that.